### PR TITLE
Moving to Dusk

### PR DESCRIPTION
**Pull request by psaunders**
### Authentication
- [x]  Page auth test
- [x] Usertypes

### Comment
- [x] Appearance

### Motion
- [x] Administrate
- [x] Appearance

### Page
- [x] Appearance

### Dashboard
- [x] Administration

### Vote
- [x] Appearance
- [x] Permissions

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -86,6 +89,6 @@
     {
       "type": "vcs",
       "url": "https://github.com/korygorsky/mailthief"
-    }
+    }
```
Comment (Kiniamaro) [inline composer.json]: trailiing whitspace

File: public/app/components/commentList/commentList.js [hunk 2]
```diff
@@ -20,10 +20,11 @@
 			self.count = utils.count;
 
 			function fetchUserCommentVotes() {
-				if (!$rootScope.authenticatedUser) {
+
+				if($rootScope.authenticatedUser === undefined || $rootScope.authenticatedUser === null || $rootScope.authenticatedUser === "" ) {
 					return false;
 				}
-
+
```
Comment (Kiniamaro) [inline public/app/components/commentList/commentList.js]: kill it with fire

File: public/app/components/login/login.svc.js [hunk 3]
```diff
@@ -12,47 +12,40 @@
...
-
+    
+    /**
+     * Clears out the logged in user and logs out
+     */
 		function clearCredentials(redirect) {
 			$rootScope.authenticatedUser = null;
 			$rootScope.userIsLoggedIn = false;
-			localStorage.clear();
+      localStorageManager.logout();
```
Comment (Kiniamaro) [inline public/app/components/login/login.svc.js]: ident seems off

File: public/app/components/login/login.svc.js [hunk 4]
```diff
@@ -68,45 +61,35 @@
...
-			authResource.register( Login.newUser ).then(successHandler, function(error) {
-				Login.creating  = false;
-				Login.authError = true;
-
-				errorHandler( error.data );
-			});
-		}
 
 		function successHandler(res) {
+      localStorageManager.remove('agreement_accepted');
```
Comment (Kiniamaro) [inline public/app/components/login/login.svc.js]: ditto

File: public/app/components/login/login.svc.js [hunk 5]
```diff
@@ -68,45 +61,35 @@
...
 
-			localStorage.setItem( 'api_token', user.api_token );
-			localStorage.setItem( 'user', JSON.stringify(user) );
-			localStorage.setItem( 'public_computer', Login.publicComputer );
       motionIndex.clear();
 
+			localStorageManager.login(user, Login.rememberMe );
+
+      // Temporary fix to the user agreement not having a service and directive
+      if(!user.agreement_accepted){
```
Comment (Kiniamaro) [inline public/app/components/login/login.svc.js]: ditto

File: public/app/components/password/resetpassword.svc.js [hunk 6]
```diff
@@ -13,20 +13,21 @@
 			'ToastMessage',
 			'utils',
 			'motionIndex',
+      'localStorageManager',
```
Comment (Kiniamaro) [inline public/app/components/password/resetpassword.svc.js]: ditto

File: public/app/components/register/register.dir.js [hunk 7]
```diff
@@ -4,27 +4,29 @@
...
+			var self = this;
+			
+			communityResource.getCommunities().then(function(results) {
 
-		return {
-			controller: function() {
-				var self = this;
-				
-				communityResource.getCommunities().then(function(results) {
+				self.communityIndex = results.data.data;
```
Comment (Kiniamaro) [inline public/app/components/register/register.dir.js]: I'm sensing this is a tab vs space thing

File: public/app/components/register/register.svc.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+		.module('iserveu')
+		.factory('registerService', [
+			'$rootScope',
+			'$timeout',
+			'authResource',
+			'ToastMessage',
+			'utils',
+			'redirectService',
+			'loginService',
+      'localStorageManager',
```
Comment (Kiniamaro) [inline public/app/components/register/register.svc.js]: it's everywhere

File: public/app/shared/authorizer/authorizer.svc.js [hunk 9]
```diff
@@ -14,10 +14,10 @@
     this.canAccess = function(requirePermissions) {
 
       var user = $rootScope.authenticatedUser;
-
+
```
Comment (Kiniamaro) [inline public/app/shared/authorizer/authorizer.svc.js]: remove this

File: public/app/shared/authorizer/authorizer.svc.js [hunk 10]
```diff
@@ -14,10 +14,10 @@
     this.canAccess = function(requirePermissions) {
 
       var user = $rootScope.authenticatedUser;
-
+      
       if (!user || angular.isUndefined(requirePermissions))
         return false;
-
+
```
Comment (Kiniamaro) [inline public/app/shared/authorizer/authorizer.svc.js]: ditto

File: public/app/shared/services/localstoragemanager.svc.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        'user',
+        'api_key',
+        'remember_me'
+        /* It does not clear agreement_accepted because that gets cleared
+           on login anyway if that user hasn't accepted it yet. Prevents the 
+           box coming up unnaturally often */
+      ];
+      
+      Object.keys(localStorage).forEach(function (key) {
+        if(clears.indexOf(key)!==-1){
```
Comment (Kiniamaro) [inline public/app/shared/services/localstoragemanager.svc.js]:
```javascript
if (clears.indexOf(key) !== -1) {

File: public/app/shared/services/localstoragemanager.svc.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+      localStorage.removeItem(name);
+
+      if(value.constructor === Array || value.constructor === Object) value = JSON.stringify(value);
+      localStorage.setItem(name, value);
+    };
+
+    this.get = function(name, defaultValue) {
+      var value = localStorage.getItem(name);
+      
+      if(value===null) return defaultValue;
```
Comment (Kiniamaro) [inline public/app/shared/services/localstoragemanager.svc.js]:
```javascript
if (value === null) return defaultValue;

File: tests/Browser/Authentication/RegisterTest.php [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+                  ->typeBetter('@lastName', $user->last_name)
+                  ->typeBetter('@newEmail', $user->email)
+                  ->typeBetter('@confirmEmail', $user->email)
+                  ->typeBetter('@newPassword', 'abcdfsadf!!!!sdf')
+                  ->press('@submitCreate')
+                  ->waitForLocation('/#/home');
+        });
+    }
+
+        /*
```
Comment (Kiniamaro) [inline tests/Browser/Authentication/RegisterTest.php]: I'd get rid of the commented stuff

File: tests/Browser/Motion/MotionSidebarTest.php [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+            $browser->visit(new SidebarSection())
+                    ->clickBetter('@filterButton')//->pause(1000)
+                    ->waitFor('@motionStatusFilterSelect')
+                    ->clickBetter('@motionStatusFilterSelect');
+
+            $closedOption = $browser->findTagContainingText('md-option', 'Closed');
+
+            $browser->clickBetter($closedOption)
+
+                    ->debug();
```
Comment (Kiniamaro) [inline tests/Browser/Motion/MotionSidebarTest.php]: remove if not needed

File: tests/Browser/Motion/MotionSidebarTest.php [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+     **/
+    public function can_filter_motions_by_different_status()
+    {
+        $this->user = static::getPermissionedUser('create-motion');
+
+        $this->motion = factory(Motion::class, 'closed')->create();
+
+        $this->browse(function (Browser $browser) {
+            $browser->visit(new SidebarSection())
+                    ->clickBetter('@filterButton')//->pause(1000)
```
Comment (Kiniamaro) [inline tests/Browser/Motion/MotionSidebarTest.php]: remove the comment

File: tests/Browser/Pages/HomePage.php [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+     *
+     * @return array
+     */
+    public function elements()
+    {
+        return [
+              '@homeIntroduction'  => 'home-introduction',
+              '@topMotions'        => 'top-motions',
+              '@topComments'       => 'top-comments',
+          '@yourVotes'             => 'my-votes',
```
Comment (Kiniamaro) [inline tests/Browser/Pages/HomePage.php]: ident seems fucked

File: tests/PolishedTest.php [hunk 17]
```diff
@@ -77,7 +80,7 @@ public function setUp()
         parent::setUp();
 
         foreach (\File::files(base_path('storage/logs')) as $filename) {
-            File::delete($filename);
+            \File::delete($filename);
```
Comment (Kiniamaro) [inline tests/PolishedTest.php]: remove the backslash

File: tests/dusktools/Browser.php [hunk 18]
```diff
@@ -0,0 +1,317 @@
...
+
+      $this->resolver = $resolver ?: new ElementResolver($driver);
+  }
+
+  /**
+   * Resolve the element for a given button by text.
+   */
+  public function findTagContainingText($tag, $text)
+  {
+      dump('here');
```
Comment (Kiniamaro) [inline tests/dusktools/Browser.php]: dump the dump

File: webpack.mix.js [hunk 19]
```diff
@@ -33,13 +33,153 @@ mix.combine([
...
 
-mix.combine([
-    'public/app/deprecated/*.js',
-    'public/app/*.js',
-    'public/app/shared/**/*.js',
-    'public/app/api/*.js',
-    'public/app/components/**/*.js',
-    'public/app/models/**/*.js'
+mix.js([
+  'public/app/app.js',
```
Comment (Kiniamaro) [inline webpack.mix.js]: can't this just be `public/app/*.js` ?

adding everything seems overkill

File: public/app/components/register/register.svc.js (hunk 8 above)
Comment (psaunders) [inline public/app/components/register/register.svc.js]: Yeah I've just turned on JS linting for the project, but its a mix of tabs and spaces in most files

File: tests/PolishedTest.php (hunk 17 above)
Comment (psaunders) [inline tests/PolishedTest.php]: That would error out because the class is namespaced.

File: webpack.mix.js (hunk 19 above)
Comment (psaunders) [inline webpack.mix.js]: No it sucks, to have it watching all files you have to explicitly state them

**Pull request merged**